tests/foreman/api/test_hostgroup.py

```diff
@@ -1,9 +1,171 @@
 
 
+class HostGroupTestCase(APITestCase):
+    """Tests for host group entity."""
+
+    def test_bz_1107708(self):
+        """@Test: Host that created from HostGroup entity with PuppetClass
+        assigned to it should inherit such puppet class information under
+        'all_puppetclasses' field
+
+        @Feature: HostGroup and Host
+
+        @Assert: Host inherited 'all_puppetclasses' details from HostGroup that
+        was used for such Host create procedure
+
+        """
+        # Creating entities like organization, content view and lifecycle_env
+        # with not utf-8 names for easier interaction with puppet environment
+        # further in test
+        org = entities.Organization(name=gen_string('alpha')).create()
+        location = entities.Location(organization=[org]).create()
+        # Creating puppet repository with puppet module assigned to it
+        product = entities.Product(organization=org).create()
+        puppet_repo = entities.Repository(
+            content_type='puppet',
+            product=product,
+            url=FAKE_0_PUPPET_REPO,
+        ).create()
+        puppet_repo.sync()
+        with open(get_data_file(PUPPET_MODULE_NTP_PUPPETLABS), 'rb') as handle:
+            puppet_repo.upload_content(handle)
+
+        content_view = entities.ContentView(
+            organization=org,
+            name=gen_string('alpha'),
+        ).create()
+        results = content_view.available_puppet_modules()['results']
+        # Working with 'ntp' module as we know for sure that it contains at
+        # least few puppet classes
+        for result in results:
+            if result['name'] == 'ntp':
+                content_view.add_puppet_module(
+                    result['author'],
+                    result['name']
+                )
+                break
+
+        content_view.publish()
+        cvv = entities.ContentViewVersion(
+            id=content_view.read_json()['versions'][0]['id']
+        )
+        lc_env = entities.LifecycleEnvironment(
+            organization=org,
+            name=gen_string('alpha')
+        ).create()
+        cvv.promote(lc_env.id)
+        cv_attrs = content_view.read_json()
+        self.assertEqual(len(cv_attrs['versions']), 1)
+        self.assertEqual(len(cv_attrs['puppet_modules']), 1)
+
+        # Form environment name variable for our test
+        env_name = 'KT_{0}_{1}_{2}_{3}'.format(
+            org.name, lc_env.name, content_view.name, str(content_view.id))
+
+        # Get all environments for current organization.
+        # We have two environments (one created after publishing and one more
+        # was created after promotion), so we need to select promoted one
+        response = client.get(
+            entities.Environment().path(),
+            auth=get_server_credentials(),
+            data={
+                u'organization_id': org.id,
+            },
+            verify=False,
+        )
```

Comment:
:+1:

I like the approach you've taken. You're reading some number of environments, so you use `Environment().path()` and then filter down the results by passing in `data={'organization_id': org.id}`. Awesome. I consider this to be a much better approach than using `/api/organizations/:organization_id/environments`.